virtool/samples/db.py

```diff
@@ -94,6 +99,22 @@
 }
 
 
+async def attach_labels(app, document):
+    if document.get("labels"):
+        for index, label_id in enumerate(document["labels"]):
+            async with AsyncSession(app["postgres"]) as session:
+                result = await session.execute(select(Label).filter_by(id=label_id))
+                label = result.scalar()
+
+            document["labels"][index] = {
+                "id": label_id,
+                "name": label.name,
+                "description": label.description,
+                "color": label.color
+            }
```

Comment:
Can this be done in a single SQL query?

---

Declaring an empty `list`, then appending each label `dict` is cleaner and less error prone than updating the old list by index. In the return statement, you can write:
```py
return {
  **document,
  "labels": labels
}
```

---

You are writing this in a number of places:
```py
{
    "id": label_id,
    "name": label.name,
    "description": label.description,
    "color": label.color
}
```

I think the idea with SQLAlchemy is that you should have a method on the `Label` class (eg. `Label.as_dict()` that does this for you. Can you look into this to confirm that I am right?